### PR TITLE
Added Consumer.store_offsets() API

### PR DESCRIPTION
**Pull request by ctrochalakis**
Hello all,

This PR implements the `Cosumer.store_offsets()` API. The signature is similar to the `.commit()` method.

### CODE REVIEW

File: confluent_kafka/src/Consumer.c [hunk 1]
```diff
@@ -325,6 +325,83 @@ static PyObject *Consumer_commit (Handle *self, PyObject *args,
...
+				     "expected %s", MessageType.tp_name);
+			return NULL;
+		}
+
+		m = (Message *)msg;
+
+		c_offsets = rd_kafka_topic_partition_list_new(1);
+		rd_kafka_topic_partition_list_add(
+			c_offsets, cfl_PyUnistr_AsUTF8(m->topic, &uo8),
+			m->partition)->offset =m->offset + 1;
```
Comment (edenhill) [inline confluent_kafka/src/Consumer.c]: nit: space after =

File: confluent_kafka/src/Consumer.c [hunk 2]
```diff
@@ -325,6 +325,83 @@ static PyObject *Consumer_commit (Handle *self, PyObject *args,
...
+		Py_XDECREF(uo8);
+	}
+
+
+	err = rd_kafka_offsets_store(self->rk, c_offsets);
+
+	if (c_offsets)
```
Comment (edenhill) [inline confluent_kafka/src/Consumer.c]: this is always true

Comment (ctrochalakis) [reply confluent_kafka/src/Consumer.c]: That's true, fixed!

File: confluent_kafka/src/Consumer.c (hunk 1 above)
Comment (ctrochalakis) [reply confluent_kafka/src/Consumer.c]: fixed

File: confluent_kafka/src/Consumer.c [hunk 3]
```diff
@@ -341,7 +341,7 @@ static PyObject *Consumer_store_offsets (Handle *self, PyObject *args,
 		     "buildtime 0x%x)",
 		     rd_kafka_version(), RD_KAFKA_VERSION);
 	return NULL;
-#endif
+#else
```
Comment (edenhill) [inline confluent_kafka/src/Consumer.c]: good catch

File: confluent_kafka/src/Consumer.c [hunk 4]
```diff
@@ -341,7 +341,7 @@ static PyObject *Consumer_store_offsets (Handle *self, PyObject *args,
 		     "buildtime 0x%x)",
 		     rd_kafka_version(), RD_KAFKA_VERSION);
 	return NULL;
-#endif
```
Comment (edenhill) [inline confluent_kafka/src/Consumer.c]: With this #else guard the variables defined above will be unused and produce warnings.
I suggest moving the variable decls down to the #else block.

**Comment by edenhill** [comment]
The thread test failure is not your fault, ignore it.
Waiting for clogged up travis osx workers for final verification.

**Comment by edenhill** [comment]
Thanks for this!

**Pull request merged**